src/lib.rs

```diff
@@ -411,6 +411,80 @@ impl<'text> BidiInfo<'text> {
         result.into()
     }
 
+    /// Returns a map from the visual order to the logical order.
+    ///
+    ///   ## Example
+    ///
+    /// ```rust
+    /// # #[cfg(feature = "hardcoded-data")] {
+    /// use unicode_bidi::BidiInfo;
+    ///
+    /// // This example text is defined using `concat!` because some browsers
+    /// // and text editors have trouble displaying bidi strings.
+    /// let text = concat![
+    ///   "א",
+    ///   "ב",
+    ///   "ג",
+    ///   "a",
+    ///   "b",
+    ///   "c",
+    /// ];
+    ///
+    /// // Resolve embedding levels within the text.  Pass `None` to detect the
+    /// // paragraph level automatically.
+    /// let bidi_info = BidiInfo::new(&text, None);
+    ///
+    /// // This paragraph has embedding level 1 because its first strong character is RTL.
+    /// assert_eq!(bidi_info.paragraphs.len(), 1);
+    /// let para = &bidi_info.paragraphs[0];
+    /// assert_eq!(para.level.number(), 1);
+    /// assert_eq!(para.level.is_rtl(), true);
+    ///
+    /// // Re-ordering is done after wrapping each paragraph into a sequence of
+    /// // lines. For this example, I'll just use a single line that spans the
+    /// // entire paragraph.
+    /// let line = para.range.clone();
+    ///
+    /// let display = bidi_info.reorder_line(para, line);
+    /// assert_eq!(display, concat![
+    ///   "a",
+    ///   "b",
+    ///   "c",
+    ///   "ג",
+    ///   "ב",
+    ///   "א",
+    /// ]);
+    ///
+    /// let line = para.range.clone();
+    /// let order = bidi_info.reorder_index(para);
+    ///
+    /// assert_eq!(order, [ 6, 7, 8, 4, 2, 0 ]);
```

Comment:
Nit: Are the indices returned relative to the start of the paragraph or the start of the whole string? Please add a test to clarify this.

Reply:
+1